utils/dynamicip/updater.go

```diff
@@ -88,11 +92,13 @@ func (u *updater) Dispatch(log logging.Logger) {
 				continue
 			}
 
-			if !newIP.Equal(oldIP) {
-				u.dynamicIP.SetIP(newIP)
+			if newAddr != oldAddr {
+				u.dynamicIP.Set(netip.AddrPortFrom(newAddr, port))
 				log.Info("updated public IP",
-					zap.Stringer("newIP", newIP),
+					zap.Stringer("oldIP", oldAddr),
+					zap.Stringer("newIP", newAddr),
 				)
+				oldAddr = newAddr
```

Comment:
That was why I coded it this way... I don't know that it really matters that much... `u.dynamicIP.Set` is the only time the write lock is expected to be held for this value... So I doubt there is any noticeable locking improvement... But looking at this function in a vacuum, this does seem to avoid locking.